saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -64,24 +73,46 @@ def create_deliveries_for_subscriptions(
 
     event_payloads = []
     event_deliveries = []
+
+    # Dataloaders are shared between calls to generate_payload_from_subscription to
+    # reuse their cache. This avoids unnecessary DB queries when different webhooks
+    # need to resolve the same data.
+    dataloaders: dict[str, type[DataLoader]] = {}
+
+    request = initialize_request(
+        requestor,
+        event_type in WebhookEventSyncType.ALL,
+        event_type=event_type,
+        allow_replica=allow_replica,
+        request_time=request_time,
+        dataloaders=dataloaders,
+    )
+
     for webhook in webhooks:
         data = generate_payload_from_subscription(
             event_type=event_type,
             subscribable_object=subscribable_object,
             subscription_query=webhook.subscription_query,
-            request=initialize_request(
-                requestor,
-                event_type in WebhookEventSyncType.ALL,
-                event_type=event_type,
-                allow_replica=allow_replica,
-            ),
+            request=request,
             app=webhook.app,
         )
+
         if not data:
             logger.info(
                 "No payload was generated with subscription for event: %s" % event_type
             )
             continue
+
+        if (
+            settings.ENABLE_LIMITING_WEBHOOKS_FOR_IDENTICAL_PAYLOADS
+            and pre_save_payloads
+        ):
+            key = get_pre_save_payload_key(webhook, subscribable_object)
+            pre_save_payload = pre_save_payloads.get(key)
+            if pre_save_payload and pre_save_payload == data:
+                logger.info("Skipping webhook: %s - no data changes.", event_type)
```

Comment:
It might be beneficial to include additional details in these logs, such as `event_type` and webhook's `target_url` (which can be helpful for debugging). Furthermore, we can consider aggregating all skipped webhooks into a single log message (with `extra` data).

Reply:
Good idea about including target URL 👍